### PR TITLE
Add new MITRE API endpoints and framework functions

### PR DESCRIPTION
**Pull request by mcarmona99**
|Related issue|
|---|
| https://github.com/wazuh/wazuh/issues/7819|

Hi team,

This PR closes https://github.com/wazuh/wazuh/issues/7819.

Due to the changes in MITRE db performed in #7607, we have adapted the framework to the changes and created new API endpoints to access all the MITRE information.

New endpoints:

- `GET /mitre/groups`
- `GET /mitre/metadata`
- `GET /mitre/mitigations`
- `GET /mitre/references`
- `GET /mitre/software`
- `GET /mitre/tactics`
- `GET /mitre/techniques`

Tests results:

```
framework/wazuh/core/tests/
============== 664 passed, 1 xfailed, 2 xpassed, 9 warnings in 4.86s ==============

framework/wazuh/tests/
============== 586 passed, 13 warnings in 62.47s (0:01:02) ==============
```

```
test_mitre_endpoints.tavern.yaml 
	 7 passed, 13 warnings

test_rbac_black_mitre_endpoints.tavern.yaml 
	 7 passed, 13 warnings

test_rbac_white_mitre_endpoints.tavern.yaml 
	 7 passed, 13 warnings
```

### CODE REVIEW

File: api/api/controllers/mitre_controller.py [hunk 1]
```diff
@@ -1,50 +1,379 @@
...
-    :param sort: Sort the items. Format: {'fields': ['field1', 'field2'], 'order': 'asc|desc'}
-    :param select: Select fields to return. Format: {"fields":["field1","field2"]}.
-    :param q: Query to filter by
-    :return: Data
+async def get_metadata(request, pretty=False, wait_for_complete=False):
+    """Return the metadata of the MITRE's database
+
+    Parameters
+    ----------
+    pretty : bool, optional
```
Comment (davidjiglesias) [inline api/api/controllers/mitre_controller.py:22]: Add request parameter also

File: api/api/spec/spec.yaml [hunk 2]
```diff
@@ -10565,79 +10626,499 @@ paths:
...
+                data:
+                  affected_items:
+                    - value: "1"
+                      key: "db_version"
+                    - value: "2.0"
+                      key: "mitre_version"
+                  total_affected_items: 2
+                  failed_items: [2]
+                  total_failed_items: 0
+                message: "Metadata information was returned"
```
Comment (davidjiglesias) [inline api/api/spec/spec.yaml:10760]: Add MITRE here and to the SDK response.

File: api/api/spec/spec.yaml [hunk 3]
```diff
@@ -10565,79 +10626,499 @@ paths:
...
+          $ref: '#/components/responses/InvalidHTTPMethodResponse'
+        '429':
+          $ref: '#/components/responses/TooManyRequestsResponse'
+
+  /mitre/mitigations:
+    get:
+      tags:
+        - MITRE
+      summary: "Get MITRE mitigations"
+      description: "Return the mitigation from MITRE database"
```
Comment (davidjiglesias) [inline api/api/spec/spec.yaml:10778]:
```suggestion
      description: "Return the mitigations from MITRE database"
```

File: framework/wazuh/core/mitre.py [hunk 4]
```diff
@@ -0,0 +1,548 @@
...
+        """
+
+        if filters is None:
+            filters = {}
+        if min_select_fields is None:
+            if table == 'phase':
+                self.min_select_fields = {'tactic_id', 'tech_id'}
+                default_sort_field = 'tactic_id'
+            elif table == 'mitigate' or table == 'use':
+                # source_id = mitigation_id or group_id or software_id, target_id = technique_id
```
Comment (davidjiglesias) [inline framework/wazuh/core/mitre.py:88]: ?

File: framework/wazuh/mitre.py [hunk 5]
```diff
@@ -1,141 +1,296 @@
...
-                    if k not in unique_fields and v not in result[found_in_pos][k]:
-                        # We append the fields to their corresponding dict if id is already present
-                        result[found_in_pos][k].append(v)
-            except IndexError:
-                # We format and add a dict to result if id was not found
-                result.append(
-                    {k: (json.loads(v) if k == 'json' else v) if k in unique_fields else [v] for k, v in entry.items()})
-        self._data = result
-        return super()._format_data_into_dictionary()
+logger = logging.getLogger('wazuh')
```
Comment (davidjiglesias) [inline framework/wazuh/mitre.py:12]: ??

File: framework/wazuh/mitre.py [hunk 6]
```diff
@@ -1,141 +1,296 @@
-# Copyright (C) 2015-2020, Wazuh Inc.
+# Copyright (C) 2015-2021, Wazuh Inc.
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 
-import json
+import logging
```
Comment (davidjiglesias) [inline framework/wazuh/mitre.py:5]: ??

File: framework/wazuh/mitre.py [hunk 7]
```diff
@@ -1,141 +1,296 @@
...
-    if id_:
-        q = f'{q};id={id_}' if q else f'id={id_}'
-    if phase_name:
-        q = f'{q};phase_name={phase_name}' if q else f'phase_name={phase_name}'
-    if platform_name:
-        q = f'{q};platform_name={platform_name}' if q else f'platform_name={platform_name}'
-
-    # Execute query
-    db_query = WazuhDBQueryMitre(offset=offset, limit=limit, query=q, sort=sort, search=search, select=select)
+    result = AffectedItemsWazuhResult(none_msg='No metadata information was returned',
```
Comment (davidjiglesias) [inline framework/wazuh/mitre.py:23]: Add MITRE to messages as specified in spec review

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: api/api/controllers/mitre_controller.py (hunk 1 above)
Comment (mcarmona99) [reply api/api/controllers/mitre_controller.py:22]: done

File: api/api/spec/spec.yaml (hunk 2 above)
Comment (mcarmona99) [reply api/api/spec/spec.yaml:10760]: Done

File: api/api/spec/spec.yaml (hunk 3 above)
Comment (mcarmona99) [reply api/api/spec/spec.yaml:10778]: done

File: framework/wazuh/mitre.py (hunk 5 above)
Comment (mcarmona99) [reply framework/wazuh/mitre.py:12]: logger left after debugging, removed

File: framework/wazuh/mitre.py (hunk 6 above)
Comment (mcarmona99) [reply framework/wazuh/mitre.py:5]: same as last comment

File: framework/wazuh/mitre.py (hunk 7 above)
Comment (mcarmona99) [reply framework/wazuh/mitre.py:23]: done

File: framework/wazuh/core/mitre.py (hunk 4 above)
Comment (mcarmona99) [reply framework/wazuh/core/mitre.py:88]: updated comment,
source_id refers to mitigation_id, group_id or software_id
target_id alwauys refers to technique_id

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**